tests/test_service.py

```diff
@@ -1,6 +1,49 @@
-"""Tests go here."""
+import json
+import os
+import pytest
+import threading, time
+from app import listen
+from kafka import KafkaProducer
 
+msg = {'test': 'aiops-qa-test'}
 
-def test_something():
-    """Test something."""
-    pass
+class Producer(threading.Thread):
+    def __init__(self, server, kafka_topic, msg):
+        threading.Thread.__init__(self)
+        self.stop_event = threading.Event()
+        self.server = server
+        self.topic = kafka_topic
+        self.msg = msg
+
+    def stop(self):
+        self.stop_event.set()
+
+    def run(self):
+        producer = KafkaProducer(bootstrap_servers=self.server)
+
+        while not self.stop_event.is_set():
+            producer.send(self.topic, json.dumps(self.msg).encode())
+        producer.close()
+
+@pytest.fixture
+def topic():
+    return os.environ.get('KAFKA_TOPIC')
+
+@pytest.fixture
+def kafka_server():
+    return os.environ.get('KAFKA_SERVER')
+
+@pytest.fixture
+def producer(kafka_server, topic):
+    producer = Producer(server=kafka_server, kafka_topic=topic, msg=msg)
+    producer.start()
```

Comment:
The fact you run `Producer` in a thread could be problematic with a yield fixture the way you are using it here. The example I'm thinking of is:
1) You call `.start()` on producer
2) Producer takes a few sec to actually "get connected" and produce a message
3) Meanwhile, since `.start()` is non-blocking, the fixture has yieled, and the test has already started running which depends on the fixture.